x-pack/test/security_api_integration/tests/session_idle/extension.ts

```diff
@@ -93,6 +99,7 @@ export default function ({ getService }: FtrProviderContext) {
       it('should extend the session', async () => {
         // browsers will follow the redirect and return the new session info, but this testing framework does not
         // we simulate that behavior in this test by sending another GET request
+        await setTimeoutAsync(200); // wait a bit
```

Comment:
__nit:__ maybe it'd make more sense to have this delay right before we try to extend session and add a comment in the code to explain why we have it (what you mentioned in the issue description does a great job explaining why we need this delay).